typegate/src/runtimes/typegraph.ts

```diff
@@ -158,11 +170,29 @@ export class TypeGraphRuntime extends Runtime {
         };
 
         visitTypes(this.tg, getChildTypes(this.tg.types[0]), myVisitor);
-
-        const scalarTypes = distinctBy(
-          [...scalarTypeIndices].map((idx) => this.tg.types[idx]),
+        const distinctScalars = distinctBy(
+          [...scalarTypeIndices].map((idx) => {
+            const node = this.tg.types[idx];
+            this.scalarIndex.set(node.type, idx);
+            return node;
+          }),
           (t) => t.type, // for scalars: one GraphQL type per `type` not `title`
-        ).map((type) => this.formatType(type, false, false));
+        );
+        const scalarTypes = distinctScalars.map((type) =>
+          this.formatType(type, false, false)
+        );
+        const customScalarTypes = distinctScalars.map((node) => {
+          const key = generateCustomScalarName(node);
+          const custom = {
+            title: key,
```

Comment:
all output fields should be lazy, e.g. `() => key`

Reply:
I don't understand, `custom` is of type `ObjectNode`
https://github.com/metatypedev/metatype/blob/main/typegate/src/types/typegraph.ts#L86